<launch>
    <node name="line_detector_node" pkg="laser_line_extraction" type="line_detector_node">
        <param name="~frequency" value="20.0" />
        <param name="~frame_id" value="base_laser" />
        <param name="~scan_topic" value="scan_filtered" />
        <param name="~publish_markers" value="true" />
        <param name="~bearing_std_dev" value="1e-5" />
        <param name="~range_std_dev" value="0.02" />
        <param name="~least_sq_angle_thresh" value="0.01" />
        <param name="~least_sq_radius_thresh" value="0.01" />
        <param name="~max_line_gap" value="0.05" />
        <param name="~min_line_length" value="0.15" />
        <param name="~min_range" value="0.1" />
        <param name="~min_split_dist" value="0.05" />
        <param name="~outlier_dist" value="0.06" />
        <param name="~min_line_points" value="5" />
        <param name="~use_scan" value="false"/>


        <param name="debug" value="false"/>
        <param name="max_range" value="2.1"/>
        <param name="min_angle" value="-1.575"/>
        <param name="max_angle" value="1.575"/>

        <param name="filter_window" value="4"/>

        <param name="min_grow_dist" value="0.02"/>
        <param name="min_gap_dist" value="0.05"/>

        <param name="min_segLength" value="0.15"/>
        <param name="max_segLength" value="0.28"/>
        <param name="min_segAngle" value="-1.8"/>
        <param name="max_segAngle" value="1.8"/>
        <param name="min_segDist" value="0.1"/>
        <param name="max_segDist" value="2.5"/>

        <param name="min_shapeAngle" value="1.74"/>
        <param name="max_shapeAngle" value="2.7"/>
        <param name="min_pairInterAngle" value="-1.8"/>
        <param name="max_pairInterAngle" value="1.8"/>
        <param name="min_pairInterLength" value="0.15"/>
        <param name="max_pairInterLength" value="0.3"/>

        <param name="max_fit_error" value="0.01"/>


        <param name="expire_sec" value="10"/>

    </node>
</launch>

